### PR TITLE
Add a config arg to just save an hf checkpoint

### PR DESCRIPTION
**Pull request by dakinggg**
We have an offline conversion script from composer to hf checkpoint format, but it only works for monolithic checkpoints, and has some other components hardcoded. We are also working on improved utilities for working with checkpoints in composer, but as a stopgap, we are introducing an arg here that will run train.py, but just call the hf checkpointer save and then exit.

Test run: `just-hf-4-nWWD05` (and confirmed the checkpoint shows up in object store)

### CODE REVIEW

File: scripts/train/train.py [hunk 1]
```diff
@@ -527,6 +527,20 @@ def main(cfg: DictConfig) -> Trainer:
...
     )
 
+    # Optionally just save an HF checkpoint
+    if train_cfg.just_hf_checkpoint:
+        hf_checkpointer_callbacks = [
+            c for c in callbacks if isinstance(c, HuggingFaceCheckpointer)
+        ]
+        if len(hf_checkpointer_callbacks) == 0:
+            raise ValueError(
+                'No HuggingFaceCheckpointer callback found, but just_hf_checkpoint was set to True.',
```
Comment (mvpatel2000) [inline scripts/train/train.py:537]:
```suggestion
                'No HuggingFaceCheckpointer callback found, but just_hf_checkpoint was set to True. Please add a HuggingFaceCheckpointer.',
```

File: llmfoundry/utils/config_utils.py [hunk 2]
```diff
@@ -161,6 +161,7 @@ class TrainConfig:
     load_strict_model_weights: bool = True
     load_ignore_keys: Optional[List[str]] = None
     save_ignore_keys: Optional[List[str]] = None
+    just_hf_checkpoint: bool = False
```
Comment (mvpatel2000) [inline llmfoundry/utils/config_utils.py:164]:
```suggestion
    only_hf_checkpoint: bool = False
```
More consistent with naming elsewhere

Comment (dakinggg) [reply llmfoundry/utils/config_utils.py]: done

**Review by mvpatel2000 — DISMISSED** [review]

File: scripts/train/train.py [hunk 3]
```diff
@@ -527,6 +527,20 @@ def main(cfg: DictConfig) -> Trainer:
...
+    # Optionally just save an HF checkpoint
+    if train_cfg.only_hf_checkpoint:
+        hf_checkpointer_callbacks = [
+            c for c in callbacks if isinstance(c, HuggingFaceCheckpointer)
+        ]
+        if len(hf_checkpointer_callbacks) == 0:
+            raise ValueError(
+                'No HuggingFaceCheckpointer callback found, but only_hf_checkpoint was set to True. Please add a HuggingFaceCheckpointer.',
+            )
+
```
Comment (mvpatel2000) [inline scripts/train/train.py:539]: should raise error or warning if > 1 hf checkpointer

Comment (dakinggg) [reply scripts/train/train.py:543]: added warning

**Review by mvpatel2000 — APPROVED** [review]

File: scripts/train/train.py [hunk 4]
```diff
@@ -527,6 +527,24 @@ def main(cfg: DictConfig) -> Trainer:
...
+            c for c in callbacks if isinstance(c, HuggingFaceCheckpointer)
+        ]
+        if len(hf_checkpointer_callbacks) == 0:
+            raise ValueError(
+                'No HuggingFaceCheckpointer callback found, but only_hf_checkpoint was set to True. Please add a HuggingFaceCheckpointer.',
+            )
+        if len(hf_checkpointer_callbacks) > 1:
+            warnings.warn(
+                'Multiple HuggingFaceCheckpointer callbacks found. Only the first one will be used.',
+            )
```
Comment (mvpatel2000) [inline scripts/train/train.py:542]: i would just raise an error

**Review by snarayan21 — APPROVED** [review]
dope

File: scripts/train/train.py (hunk 4 above)
Comment (dakinggg) [reply scripts/train/train.py]: sure

**Pull request merged**